magni_bringup/scripts/launch_core.py

```diff
@@ -68,10 +70,26 @@ def get_conf():
         print "Error calling pifi"
         subprocess.call(["chronyc", "waitsync", "6"]) # Wait up to 60 seconds for chrony
 
+    boardRev = 0
+
+    if conf['motor_controller']['board_version'] == None: 
+        # Code to read board version from I2C
+        # The I2C chip is only present on 5.0 and newer boards
+        try:
+            i2cbus = smbus.SMBus(1)
+            i2cbus.write_byte(0x20, 0xFF)
+            time.sleep(0.2)
+            inputPortBits = i2cbus.read_byte(0x20)
+            boardRev = 49 + (15 - (inputPortBits & 0x0F))
+            print "Got board rev: %d" % boardRev
+        except: 
+            print "Error reading board version from i2c"
```

Comment:
Should we say  'Error reading motor controller board version from i2c'  here?   You decide.